partner_company_type/models/res_partner.py

```diff
@@ -10,3 +11,33 @@ class ResPartner(models.Model):
     partner_company_type_id = fields.Many2one(
         comodel_name="res.partner.company.type", string="Legal Form"
     )
+
+    @api.constrains(
+        "country_id",
+        "state_id",
+        "partner_company_type_id",
+    )
+    def _check_partner_company_type_country_state(self):
+        for rec in self:
+            if not rec.partner_company_type_id:
+                return
+            if (
+                rec.partner_company_type_id.country_ids
+                and rec.country_id not in rec.partner_company_type_id.country_ids
+            ):
+                raise UserError(
```

Comment:
You should raise `ValidationError`in a contraint.

Reply:
@rousseldenis corrected :)